src/main/java/net/sf/jabref/exporter/SavePreferences.java

```diff
@@ -51,7 +51,7 @@ public static SavePreferences loadForExportFromPreferences(JabRefPreferences pre
     }
 
     public static SavePreferences loadForSaveFromPreferences(JabRefPreferences preferences) {
-        Boolean saveInOriginalOrder = true;
```

Comment:
Yes `saveInOriginalOrder = true` let JabRef ignore the save order specified in the metadata. It was a bug that this was set for the save procedure (fixed in this PR). So now the metadata is taken into account for saves. For exports, user can choose to always export in the original order (see Preference dialog) regardless of what was specified in the metadata (and this preference option is the whole purpose of the boolean variable).